Adding ChatGPT as macOS Fleet-maintained app

harrisonravazzolo opened the pull request:
<!-- Add the related story/sub-task/bug number, like Resolves #123, or remove if NA -->
**Related issue:** Resolves #34998 

# Checklist for submitter

If some of the following don't apply, delete the relevant line.

- [ ] Changes file added for user-visible changes in `changes/`, `orbit/changes/` or `ee/fleetd-chrome/changes`.
  See [Changes files](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/guides/committing-changes.md#changes-files) for more information.

- [ ] Input data is properly validated, `SELECT *` is avoided, SQL injection is prevented (using placeholders for values in statements)
- [ ] If paths of existing endpoints are modified without backwards compatibility, checked the frontend/CLI for any necessary changes

## Testing

- [ ] Added/updated automated tests
- [ ] Where appropriate, [automated tests simulate multiple hosts and test for host isolation](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/reference/patterns-backend.md#unit-testing) (updates to one hosts's records do not affect another)

- [ ] QA'd all new/changed functionality manually

For unreleased bug fixes in a release candidate, one of:

- [ ] Confirmed that the fix is not expected to adversely impact load test results
- [ ] Alerted the release DRI if additional load testing is needed

## Database migrations

- [ ] Checked schema for all modified table for columns that will auto-update timestamps during migration.
- [ ] Confirmed that updating the timestamps is acceptable, and will not cause unwanted side effects.
- [ ] Ensured the correct collation is explicitly set for character columns (`COLLATE utf8mb4_unicode_ci`).

## New Fleet configuration settings

- [ ] Setting(s) is/are explicitly excluded from GitOps

If you didn't check the box above, follow this checklist for GitOps-enabled settings:

- [ ] Verified that the setting is exported via `fleetctl generate-gitops`
- [ ] Verified the setting is documented in a separate PR to [the GitOps documentation](https://github.com/fleetdm/fleet/blob/main/docs/Configuration/yaml-files.md#L485)
- [ ] Verified that the setting is cleared on the server if it is not supplied in a YAML file (or that it is documented as being optional)
- [ ] Verified that any relevant UI is disabled when GitOps mode is enabled

## fleetd/orbit/Fleet Desktop

- [ ] Verified compatibility with the latest released version of Fleet (see [Must rule](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/workflows/fleetd-development-and-release-strategy.md))
- [ ] If the change applies to only one platform, confirmed that `runtime.GOOS` is used as needed to isolate changes
- [ ] Verified that fleetd runs on macOS, Linux and Windows
- [ ] Verified auto-update works from the released version of component to the new version (see [tools/tuf/test](../tools/tuf/test/README.md))

harrisonravazzolo committed:
adding chatgpt as FMA

allenhouchins committed:
Update apps.json

allenhouchins commented:
Related: https://github.com/fleetdm/confidential/issues/12844

allenhouchins committed:
Rename chatgpt.json to chatgpt-desktop.json

Renamed the Homebrew input file for ChatGPT to clarify that it refers to the desktop version.

allenhouchins committed:
Rename ChatGPT Desktop to chatgpt in configs

Renamed files and updated slugs from 'chatgpt-desktop' to 'chatgpt' for consistency across input and output configuration files.

allenhouchins committed:
Add DMG extraction to ChatGPT macOS install script

The install script for ChatGPT on macOS now extracts the DMG contents before copying the app to the Applications folder. This ensures proper installation from DMG packages.

allenhouchins committed:
Add ChatGPT icon component and image asset

Introduces a new ChatGpt.tsx React component for displaying the ChatGPT icon using an embedded base64 PNG image. Also adds the corresponding app icon image asset to the project.

allenhouchins commented:
Converting back to draft. Need to reference the custom install script differently. More info here: https://fleetdm.slack.com/archives/C086V2QK76X/p1762446434515619?thread_ts=1762442675.529329&cid=C086V2QK76X

allenhouchins committed:
Add install script for ChatGPT Homebrew app

Introduces chatgpt-install.sh to automate installation of ChatGPT via Homebrew, including quitting the app if running and handling backup of existing installations. Updates chatgpt.json to reference the new install script.

allenhouchins committed:
Merge branch 'main' into fma-chatgpt

allenhouchins committed:
Merge branch 'main' into fma-chatgpt

allenhouchins commented:
All checks will pass once this is merged: https://github.com/fleetdm/fleet/pull/35390

allenhouchins committed:
Merge branch 'main' into fma-chatgpt

allenhouchins committed:
Merge branch 'main' into fma-chatgpt

allenhouchins commented:
Converting back to draft until the `pkg` vs `dmg` issue is resolved and this re-passes checks.

harrisonravazzolo committed:
Change installer format from pkg to dmg

allenhouchins committed:
updates
